icu4j/pom.xml

```diff
@@ -328,6 +331,59 @@
       </build>
     </profile>
 
+    <!-- This is used to gather API data used to generate API reports
+      (regular, draft, deprecated, status consistency, etc.)
+      To run it:
+        mvn install -DskipITs -DskipTests
+        mvn site -DskipITs -DskipTests -P gatherapi
+    -->
+    <profile>
+      <id>gatherapi</id>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>aggregate</id>
+                <phase>site</phase>
+                <goals>
+                  <goal>aggregate</goal>
+                </goals>
+              </execution>
+            </executions>
+            <configuration>
+              <!-- For incomplete doc tags javadoc fails with:
+                  ```
+                  Error while creating javadoc report:
+                  Exit code: 1
+                  Command line was: ...
+                  Refer to the generated Javadoc files in '...' dir.
+                  ```
+                But in most cases these are not related to what we do here, which is API reporting.
+                We ignore them so that we can progess with what we really care.
+              -->
+              <failOnError>false</failOnError>
```

Comment:
@yumaoka sound ok?

Reply:
I prefer to let it fail. `gatherapi` is a part of release process producing official API signature for a release. And it will be used for future API change report.

I always thought the check is done when any API changes are made. I'm not sure `gatherapi` is triggered during regular javadoc generation. But for all downstream release process, we need to fix issues reported by gatherapi - e.g. lacking docs for default no-args constructor, etc.

Reply:
TLDR: this works as expected.

The way "gatherapi" works is that com.ibm.icu.dev.tool.docs.GatherAPIData is a docket.
It completely replaces the standard javadoc doclet.
It is invoked for each class and method, but instead of collecting the comments and generating javadoc, it only adds the method names to the API report.

See https://docs.oracle.com/javase/8/docs/technotes/guides/javadoc/doclet/overview.html

So the failure is because the maven-javadoc-plugin does not detect the generated javadoc
And it does not detect it because it does not exist, as expected (it is not generated).

This didn't fail before because the ant script invoked the javadoc generation with the custom doclet,
but didn't check that the javadoc was generated, the way maven does.

The real javadoc is generated in another maven profile, which uses the standard javadoc doclet.